pygfx/objects/_base.py

```diff
@@ -353,13 +353,15 @@ def traverse(self, callback, skip_invisible=False):
             callback(child)
 
     def iter(self, filter_fn=None, skip_invisible=False):
+        """Create a generator that iterates over this objects and its children.
+        If ``filter_fn`` is given, only objects for which it returns ``True``
+        are included.
+        """
         if skip_invisible and not self.visible:
             return
 
-        if filter_fn is not None and not filter_fn(self):
-            return
-
-        yield self
+        if not (filter_fn is not None and not filter_fn(self)):
```

Comment:
```suggestion
        if filter_fn is None:
            yield self
        elif filter_fn(self):
            yield self
```

Could we write it like this? I tripped over this on my first pass, thinking that it would not yield when `filter_fn` is None. I think this makes it a little easier to read.

Reply:
I agree that that `if`'s double negations are a bit confusing. PR's merged now though. Perhaps we can sneak your suggestion via another PR that touches related code 😃

Reply:
Yes, my bad😅 I'll keep it in mind and will add it in a different PR.